vars/runIntegTestScript.groovy

```diff
@@ -15,15 +15,15 @@ void call(Map args = [:]) {
     String architecture = buildManifest.getArtifactArchitecture()
     String distribution = buildManifest.getDistribution()
     String platform = buildManifest.getArtifactPlatform()
+    String filename = buildManifest.build.getFilename()
     echo "Start integTest on: " + distribution + " " + architecture + " " + platform
 
-    def javaVersion = (jobName.equals('distribution-build-opensearch')) ? detectTestDockerAgent(testManifest: args.testManifest).javaVersion : ''
-    String javaHomeCommand = (jobName.equals('distribution-build-opensearch') && ! javaVersion.equals('')) ? "JAVA_HOME=/opt/java/${javaVersion}" : ''
-    if (platform.equals('windows')) { // Windows use scoop to switch the Java Version
+    def javaVersion = (filename == 'opensearch') ? detectTestDockerAgent(testManifest: args.testManifest).javaVersion : ''
+    String javaHomeCommand = (javaVersion != '' && platform != 'windows') ? "JAVA_HOME=/opt/java/${javaVersion}" : ''
```

Comment:
Can we do like 
```
String javaHomeCommand = (javaVersion == '' || platform == 'windows') ? "" : "JAVA_HOME=/opt/java/${javaVersion}"
```

This may be easier to understand.

Reply:
Good catch!

Reply:
I think I will just keep this logic for now as we will refactor it again.
As of now it matches the windows logic so it might be easier for the time being.